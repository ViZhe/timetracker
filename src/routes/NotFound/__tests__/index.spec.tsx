
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Link } from 'react-router-dom';

import { NotFound } from '../index';


describe('<NotFound />', () => {
  const notFound: ShallowWrapper = shallow(<NotFound />);

  it('should match snapshot', () => {
    expect(notFound).toMatchSnapshot();
  });

  it('should have one Link', () => {
    expect(notFound.find(Link).length).toEqual(1);
  });

  it('should have Link to root', () => {
    expect(notFound.find(Link).prop('to')).toEqual('/');
  });
});
