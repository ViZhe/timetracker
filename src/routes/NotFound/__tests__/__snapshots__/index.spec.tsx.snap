// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should match snapshot 1`] = `
<div
  className="box"
>
  <div
    className="header"
  >
    Oops, hmmm!
  </div>
  <div
    className="body"
  >
    Something went wrong.
  </div>
  <div
    className="footer"
  >
    <Link
      replace={false}
      to="/"
    >
      <Button
        block={false}
        ghost={false}
        icon="home"
        loading={false}
        size="large"
        type="primary"
      >
        Home
      </Button>
    </Link>
  </div>
</div>
`;
